src/tracing/types.rs

```diff
@@ -707,7 +709,7 @@ impl CallTraceStep {
             gas_cost: self.gas_cost,
             op: self.op.to_string(),
             pc: self.pc as u64,
-            refund_counter: (self.gas_refund_counter > 0).then_some(self.gas_refund_counter),
+            refund_counter: (self.gas_refund_counter > 0).then_some(self.gas_refund_counter as u64),
```

Comment:
I think we can do it like this but only if we update the refund counters accordingly, because we're now only tracking the refund on a per call level, so what we need to di is one post processing step that updates the counter when a call ended.


I believe we could do this when we initialize the first calltrace using the sum of all gas refunds up until this point.